ui/pages/settings/flask/snaps-list-tab/view-snap/view-snap.container.js

```diff
@@ -0,0 +1,58 @@
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { getSubjectsWithPermission } from '../../../../../selectors';
+import {
+  disableSnap,
+  enableSnap,
+  removePermissionsFor,
+} from '../../../../../store/actions';
+import ViewSnap from './view-snap.component';
+
+const mapStateToProps = (state, ownProps) => {
+  const { location, history } = ownProps;
+  const { pathname } = location;
+  const pathNameTail = pathname.match(/[^/]+$/u)[0];
+  const snap = state.metamask.snaps
+    ? Object.entries(state.metamask.snaps)
+        .map(([_, snapState]) => snapState)
+        .find((snapState) => {
+          const decoded = decodeURIComponent(escape(window.atob(pathNameTail)));
+          return snapState.id === decoded;
+        })
+    : undefined;
+  const connectedSubjects = getSubjectsWithPermission(
+    state,
+    snap.permissionName,
+  );
+  return {
+    snap,
+    history,
+    connectedSubjects,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    onDisconnect: (connectedOrigin, snapPermissionName) => {
+      dispatch(
+        removePermissionsFor({
```

Comment:
Lots of other examples here of calling `dispatch` on things that aren't action creators. `dispatch` isn't needed in any of these cases. These can be called directly from the component actually, rather than passed as props via `mapDispatchToProps`.

Reply:
I fixed the others, namely, `enableSnap` and `disableSnap`. Wasn't sure on changing `removePermissionsFor`, doesn't that use `dispatch`?

Reply:
On second thought, I just decided to play it safe and revert the changes taking out `forceMetamaskUpdateState`.